cpp/src/rolling/grouped_rolling.cu

```diff
@@ -269,6 +270,69 @@ __device__ T subtract_safe(T const& value, T const& delta)
            : cuda::std::numeric_limits<T>::min();
 }
 
+/**
+ * @brief For a specified idx, find the lowest value of the (sorted) orderby column that
+ * participates in a range-window query.
+ */
+template <typename ElementT, typename ElementIter>
+__device__ ElementT compute_lowest_in_window(ElementIter orderby_iter,
+                                             size_type idx,
+                                             ElementT delta)
+{
+  return subtract_safe(orderby_iter[idx], delta);
+}
+
+/**
+ * @brief For a specified idx, find the highest value of the (sorted) orderby column that
+ * participates in a range-window query.
+ */
+template <typename ElementT, typename ElementIter>
+__device__ ElementT compute_highest_in_window(ElementIter orderby_iter,
+                                              size_type idx,
+                                              ElementT delta)
+{
+  return add_safe(orderby_iter[idx], delta);
+}
+
+/**
+ * Accessor for values in an order-by column, on the device.
+ */
+template <typename T>
+struct device_value_accessor {
+  column_device_view const col;  ///< column view of column in device
+
+  /**
+   * @brief constructor
+   *
+   * @param[in] _col column device view of cudf column
+   */
+  __device__ device_value_accessor(column_device_view const& _col) : col{_col}
```

Comment:
Typical convention for using `_`:
 * Use `_var` style for private class/struct members.
 * Use `var_` or `var` style for function parameters.
 * Don't use `_` for public class/struct members.

```suggestion
  __device__ device_value_accessor(column_device_view const& col_) : col{col_}
```

Reply:
if `col` is made private then:
```suggestion
...
private:
column_device_view const _col;
....
  __device__ device_value_accessor(column_device_view const& col) : _col{col}
```

Reply:
Thanks. I've switched to using a `col_` for method parameter.